File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPr = prices[0];
        int maxProf = 0;

        for (int p:prices) {
            minPr = min(minPr, p);
            maxProf = max(maxProf, p-minPr);
        }

        return maxProf;
    }
};

File: Data Structures & Algorithms/jump-game-vii/submission-4.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        if s[-1] == '1':
            return False
            
        n = len(s)

        farthest = 0
        canJump = [False]*len(s)
        canJump[0] = True

        for i in range(n):
            print("i:", i)
            print(farthest)
            if canJump[i]:
                for j in range(max(farthest, i+minJump), min(i+maxJump+1, n)):
                    if j == n-1:
                        return True
                    if s[j] == '0':
                        canJump[j] = True
                    farthest = j+1
        
        return False
